Assets/Scripts/NPC/Grid.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Grid : MonoBehaviour
+{
+    [SerializeField] private List<Waypoint> waypoints = new List<Waypoint>();
+
+    public void SubscribeToGrid(Waypoint waypoint)
+    {
+        if (waypoints.Contains(waypoint)) return;
+        waypoints.Add(waypoint);
```

Comment:
Je zou dit ook zo kunnen typen, kan je zo laten als je wilt.
```suggestion
        if (!waypoints.Contains(waypoint)) waypoints.Add(waypoint);
```